src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.netfx.cs

```diff
@@ -138,27 +136,35 @@ internal void CreatePhysicalSNIHandle(
                 ipPreference, cachedDNSInfo, hostNameInCertificate);
         }
 
-        internal bool IsPacketEmpty(PacketHandle readPacket) => readPacket == default;
+        internal bool IsPacketEmpty(PacketHandle readPacket)
```

Comment:
It seems to me this is something that should be moved into the packet handle itself?

Reply:
In principle, yes - but this is another abstract method which we merge in #3355, with implementations which vary between TdsParserStateObjectNative and TdsParserStateObjectManaged. The assertions verify that we're not taking a PacketHandle populated for the native SNI and passed it to the managed SNI (and vice versa.)

Reply:
Hmm, ok... I guess the change I'm suggesting would make more sense if the packet handles were typed native and managed. But as you said above, we can't really do that today.